### PR TITLE
Fix 64 bit issues

### PR DESCRIPTION
**Pull request by tkreuzer**

### CODE REVIEW

File: boot/freeldr/freeldr/arch/i386/drivemap.c [hunk 1]
```diff
@@ -172,7 +172,7 @@ UCHAR DriveMapGetBiosDriveNumber(PCSTR DeviceName)
 
 VOID DriveMapInstallInt13Handler(PDRIVE_MAP_LIST DriveMap)
 {
-    ULONG*  RealModeIVT = (ULONG*)0x00000000;
+    ULONG*  RealModeIVT = (ULONG*)(ULONG_PTR)0x00000000;
```
Comment (HBelusca) [inline boot/freeldr/freeldr/arch/i386/drivemap.c]: Please move the freeldr changes into a separate commit! :)

File: dll/win32/advapi32/service/eventlog.c [hunk 2]
```diff
@@ -655,7 +655,7 @@ ElfChangeNotify(IN HANDLE hEventLog,
 
     RpcTryExcept
     {
-        Status = ElfrChangeNotify(hEventLog, RpcClientId, (DWORD)hEvent);
+        Status = ElfrChangeNotify(hEventLog, RpcClientId, (DWORD_PTR)hEvent);
```
Comment (HBelusca) [inline dll/win32/advapi32/service/eventlog.c]: Be careful because https://git.reactos.org/?p=reactos.git;a=blob;f=sdk/include/reactos/idl/eventlogrpc.idl;hb=08d30403777c2db122f2292657a46f42d9f59aa2#l87 indicates a ULONG Event, so either that or the other file is wrong.

File: boot/freeldr/freeldr/arch/i386/drivemap.c [hunk 3]
```diff
@@ -172,7 +172,7 @@ UCHAR DriveMapGetBiosDriveNumber(PCSTR DeviceName)
...
     USHORT* BiosLowMemorySize = (USHORT*)0x00000413;
```
Comment (ThFabba) [inline boot/freeldr/freeldr/arch/i386/drivemap.c]: Why not this one?

File: dll/3rdparty/libtiff/tif_win32.c [hunk 4]
```diff
@@ -243,7 +243,7 @@ TIFFFdOpen(int ifd, const char* name, const char* mode)
 			break;
 		}
 	}
-	tif = TIFFClientOpen(name, mode, (thandle_t)ifd, /* FIXME: WIN64 cast to pointer warning */
+	tif = TIFFClientOpen(name, mode, (thandle_t)(uintptr_t)ifd, /* FIXME: WIN64 cast to pointer warning */
```
Comment (ThFabba) [inline dll/3rdparty/libtiff/tif_win32.c]: This is 3rd party code, let's just disable the warning here.

File: dll/directx/wine/d3dcompiler_43/reflection.c [hunk 5]
```diff
@@ -1107,13 +1107,13 @@ static HRESULT d3dcompiler_parse_stat(struct d3dcompiler_shader_reflection *r, c
 
     skip_dword_unknown(&ptr, 1);
 
-    read_dword(&ptr, &r->input_primitive);
+    read_dword(&ptr, (DWORD*)&r->input_primitive);
```
Comment (ThFabba) [inline dll/directx/wine/d3dcompiler_43/reflection.c]: These aren't 64-bit-specific, are they? (also, this is Wine code, so perhaps disable the warning instead)
Same for windowscodecs.

File: dll/directx/wine/wined3d/context.c [hunk 6]
```diff
@@ -5708,7 +5708,7 @@ void context_draw_shaded_quad(struct wined3d_context *context, struct wined3d_te
         GL_EXTCALL(glBufferData(GL_ARRAY_BUFFER, sizeof(quad), quad, GL_STREAM_DRAW));
         GL_EXTCALL(glVertexAttribPointer(0, 2, GL_FLOAT, FALSE, sizeof(*quad), NULL));
         GL_EXTCALL(glVertexAttribPointer(1, 3, GL_FLOAT, FALSE, sizeof(*quad),
-                (void *)FIELD_OFFSET(struct blit_vertex, texcoord)));
+                UlongToPtr(FIELD_OFFSET(struct blit_vertex, texcoord))));
```
Comment (ThFabba) [inline dll/directx/wine/wined3d/context.c]: Wine code. Disable the warning?
Same for clusapi, crypt32, mshtml, msi, msrle32, gdiplus_winetest

Comment (HBelusca) [reply dll/directx/wine/wined3d/context.c]: Sending patch upstream too?

File: boot/freeldr/freeldr/arch/i386/drivemap.c (hunk 1 above)
Comment (tkreuzer) [reply boot/freeldr/freeldr/arch/i386/drivemap.c]: done

File: dll/win32/advapi32/service/eventlog.c (hunk 2 above)
Comment (tkreuzer) [reply dll/win32/advapi32/service/eventlog.c]: Fixed by @hpoussin

File: boot/freeldr/freeldr/arch/i386/drivemap.c (hunk 3 above)
Comment (tkreuzer) [reply boot/freeldr/freeldr/arch/i386/drivemap.c]: Fixed

File: dll/directx/wine/wined3d/context.c (hunk 6 above)
Comment (tkreuzer) [reply dll/directx/wine/wined3d/context.c]: I made them non-fatal for now.
